qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/dump/NavigationViewApiDumpInterceptor.kt

```diff
@@ -0,0 +1,36 @@
+package com.mapbox.navigation.qa_test_app.dump
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.formatter.UnitType
+import com.mapbox.navigation.core.internal.dump.MapboxDumpCommand
+import com.mapbox.navigation.core.internal.dump.MapboxDumpInterceptor
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.dropin.NavigationViewApi
+import java.io.FileDescriptor
+import java.io.PrintWriter
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class NavigationViewApiDumpInterceptor  constructor(
+    private val navigationViewApi: NavigationViewApi
+) : MapboxDumpInterceptor {
+
+    override fun command(): MapboxDumpCommand = MapboxDumpCommand(
+        command = "drop_in_ui",
+        listOf(
+            MapboxDumpCommand("start_free_drive"),
+            MapboxDumpCommand("start_active_guidance")
+        )
+    )
+
+    override fun description(): String = "Allows you to update navigation view state"
+
+    override fun intercept(
+        fileDescriptor: FileDescriptor,
+        writer: PrintWriter,
+        mapboxDumpCommand: MapboxDumpCommand
+    ) {
+        when (mapboxDumpCommand.toString()) {
+            "drop_in_ui:start_free_drive" -> navigationViewApi.startFreeDrive()
```

Comment:
Shouldn't the check be more strict? I mean shouldn't we check the command name and the arguments separately?

Reply:
yeah this is place holder :). not done yet

Reply:
Ok it's updated